roles/openshift_facts/library/openshift_facts.py

```diff
@@ -403,16 +403,18 @@ def set_master_selectors(facts):
             facts if they were not already present
 
     """
+    deployment_type = facts['common']['deployment_type']
+    if deployment_type == 'online':
+        selector = "type=infra"
+    else:
+        selector = "region=infra"
+
+    if 'router' in facts:
+        if 'selector' not in facts['router']:
+            facts['router']['selector'] = selector
```

Comment:
There should probably be a migration to move any existing value of openshift.master.router_selector to openshift.router.selector as well.